Util/Profiler/TokenFinder.php

```diff
@@ -40,6 +40,10 @@ public function __construct(Profiler $profiler, $defaultLimit)
      */
     public function find($ip = null, $url = null, $limit = null, $method = null, $start = null, $end = null)
     {
+        if ($this->profiler === null) {
+            return [];
+        }
```

Comment:
Why do we need to add this for Symfony 6 compatibility?

Reply:
I had to modify the signature of the constructor to allow null if the Profiler service doesn't exists and can't be injected. 
Without this change in "Prod" env we receive the following error:
```
Fatal error: Uncaught Symfony\Component\DependencyInjection\Exception\RuntimeException: Cannot autowire service "lexik_translation.token_finder": argument "$profiler" of method "Lexik\Bundle\TranslationBundle\Util\Profiler\TokenFinder::__construct()" references class "Symfony\Component\HttpKernel\Profiler\Profiler" but no such service exists. in \vendor\symfony\dependency-injection\Compiler\DefinitionErrorExceptionPass.php:54 Stack trace: #0 \vendor\symfony\dependency-injection\Compiler\AbstractRecursivePass.php(80): Symfony\Component\DependencyInjection\Compiler\DefinitionErrorExceptionPass->processValue(Object(Symfony\Component\DependencyInjection\Definition), true) #1 \vendor\symfony\dependency-injection\Compiler\DefinitionErrorExceptionPass.php(32): Symfony\Component\DependencyInjection\Compiler\AbstractRecursivePass->processValue(Array, true) #2 \vendor\symfony\dependency-injection\Compiler\AbstractRecursivePass.php(47): Symfony\Component\DependencyInjection\Compiler\DefinitionErrorExceptionPass->processValue(Array, true) #3 \vendor\symfony\dependency-injection\Compiler\Compiler.php(73): Symfony\Component\DependencyInjection\Compiler\AbstractRecursivePass->process(Object(Symfony\Component\DependencyInjection\ContainerBuilder)) #4 \vendor\symfony\dependency-injection\ContainerBuilder.php(716): Symfony\Component\DependencyInjection\Compiler\Compiler->compile(Object(Symfony\Component\DependencyInjection\ContainerBuilder)) #5 \vendor\symfony\http-kernel\Kernel.php(540): Symfony\Component\DependencyInjection\ContainerBuilder->compile() #6 \vendor\symfony\http-kernel\Kernel.php(769): Symfony\Component\HttpKernel\Kernel->initializeContainer() #7 \vendor\symfony\http-kernel\Kernel.php(190): Symfony\Component\HttpKernel\Kernel->preBoot() #8 \public\app.php(20): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request)) #9 {main} thrown in \vendor\symfony\dependency-injection\Compiler\DefinitionErrorExceptionPass.php on line 54
```

In find method we just need to take care that the profiler can be null.

Reply:
So how this can work before?

Reply:
Good question. I must admit I didn't had lot of time to analyse it. I will try to do it as soon as possible.

Reply:
@Seb I could find a bit of time today to check what is the issue here. In fact the ``TokenFinder`` service is only created when ``$config['dev_tools']['enable']`` is defined to ``true``:

In LexikTranslationExtension:
```php
if (true === $config['dev_tools']['enable']) {
	$this->buildDevServicesDefinition($container);
}
```

In which tokenFinderDefinition is created. So The Profiler is provided

```php
protected function buildDevServicesDefinition(ContainerBuilder $container)
{
	$container
		->getDefinition('lexik_translation.data_grid.request_handler')
		->addMethodCall('setProfiler', array(new Reference('profiler')));

	$tokenFinderDefinition = new Definition();
	$tokenFinderDefinition->setClass($container->getParameter('lexik_translation.token_finder.class'));
	$tokenFinderDefinition->setArguments(array(
		new Reference('profiler'),
		new Parameter('lexik_translation.token_finder.limit'),
	));

	$container->setDefinition('lexik_translation.token_finder', $tokenFinderDefinition);
}
```

So I reverted the declaration of ``TokenFinder`` (autoconfigured & autowired) service in my last commit and I provide the service as optional in ``TranslationController``. Now, like on master, the ``TokenFinder`` service is always created with the profiler instance.